NDTensors/src/abstractarray/fill.jl

```diff
@@ -1,13 +1,38 @@
-using .TypeParameterAccessors: unwrap_array_type, specify_default_type_parameters
+using .TypeParameterAccessors:
+  unwrap_array_type, specify_default_type_parameters, type_parameter
+
+## Warning to use these functions it is necessary to define `TypeParameterAccessors.position(::Type{<:YourArrayType}, ::typeof(ndims)))`
+## Creating these functions to help flatten tuples e.g. generic_randn(Array, (2,3)) -> generic_randn(Array, 2,3)
+## So that its possible to take the length of dims and check it agains the given arraytype ndims
+function generic_randn(
+  arraytype::Type{<:AbstractArray}, dims::Tuple; rng=Random.default_rng()
+)
+  return generic_randn(arraytype, dims...; rng=rng)
+end
 
 function generic_randn(arraytype::Type{<:AbstractArray}, dims...; rng=Random.default_rng())
-  arraytype_specified = specify_default_type_parameters(unwrap_array_type(arraytype))
+  arraytype_specified = specify_type_parameter(
+    unwrap_array_type(arraytype), ndims, length(dims)
+  )
+  arraytype_specified = specify_default_type_parameters(arraytype_specified)
+  @assert length(dims) == type_parameter(arraytype_specified, ndims)
   data = similar(arraytype_specified, dims...)
   return randn!(rng, data)
 end
 
+function generic_zeros(arraytype::Type{<:AbstractArray}, dims::Tuple)
+  return generic_zeros(arraytype, dims...)
+end
+
 function generic_zeros(arraytype::Type{<:AbstractArray}, dims...)
-  arraytype_specified = specify_default_type_parameters(unwrap_array_type(arraytype))
+  ## Right now I have to call specify first because its not possible to 
+  ## call specify_default_type_parameter and overwrite a default parameter.
+  ## So if arraytypes ndim is unspecified, specify it to the length of dims...
```

Comment:
I understand why you designed the code this way but I don't understand this comment.

Reply:
Sorry this was a comment reminder for me to make one on github to ask you about. 
What I mean is right now I don't believe that there is a function which can fill parameters of a struct with both default and non-default values. I was going to ask if you believe this would be a useful function to create (most likely in a subsequent PR)

Reply:
That's a good question. We could do that by defining a `DefaultTypeParamer()` object (if we don't have one already), and have an interface like `specify_type_parameters(Array, (eltype, ndims), (DefaultTypeParamer(), 3)`.

Reply:
This makes sense and I think would be a good idea! I will add this comment to the google doc to implement in a subsequent PR. Thanks!